File: find_primes.php
<?php
function isPrime($num) {
  if ($num <= 1) {
    return false;
  }
  for ($i = 2; $i <= $num / 2; $i++) {
    if ($num % $i == 0) {
      return false;
    }
  }
  return true;
}

function findPrime($num) {
  $largestPrime = 0;
  for ($j = 2; $j <= $num; $j++) {
    if (isPrime($j)) {
      $largestPrime = $j;
    }
  }

  return $largestPrime;
}


$now = time();
$prime = findPrime(500000);
echo("prime = ".$prime);
echo("\r\n");
echo("elapsed: ".(time() - $now));
?>

File: find_primes.py
import time

def isPrime(num :int):
  i: int
  max = int(num/2)
  if (num <= 1):
    return False
  for i in range(2,max):
    if num % i == 0:
      return False
  return True


def findPrime(num):
  largestPrime = 0
  for j in range(2,num): 
    if (isPrime(j)):
      largestPrime = j
  return largestPrime

now = time.time()
print("prime = " + float(findPrime(500000)).__str__())
print("elapsed: " + (time.time()- now).__str__())
